Vivekananda_session2_4VP20CS062_Prajwal/EmpManagement/employees/views.py

```diff
@@ -0,0 +1,78 @@
+from django.views import View
+from .models import Employee
+from django.http import JsonResponse
+from django.views.decorators.csrf import csrf_exempt
+from django.utils.decorators import method_decorator
+import json
+
+
+
+
+@method_decorator(csrf_exempt, name='dispatch')
```

Comment:
@Prajwal7Amin why 'dispatch' is used here?

Reply:
@anushds dispatch is used here as entry point for the incoming requests to the view and  to examine and call the get,put,post,delete methods.

Reply:
@Prajwal7Amin you're somewhat right but there's something missing in your answer.
Do you want to give another try at it?

Reply:
@anushds In this case 'dispatch' is excluded from CSRF protection.
So any incoming request will no longer be required to include a CSRF token in the request data

Reply:
> @anushds In this case 'dispatch' is excluded from CSRF protection. So any incoming request will no longer be required to include a CSRF token in the request data

This actually indicates the purpose of `exempt_csrf` decorator.
Whenever there's a incoming request, you'd want to know what's the HTTP method of the request (GET,POST etc.)  and then you'll have a elif ladder which would execute the appropriate logic depending on the HTTP method.
For example:
`
if request==GET:
   do something...
elif request==POST:
    do something...
`
That's where the 'dispatch` functionality comes in.
With this, you can write a class based view and have the instance methods of that class as the HTTP methods.
It will internally execute the appropriate method depending on the HTTP request (like you've defined in the code).

Reply:
@anushds Thanks for the information 👍.